cirq-rigetti/cirq_rigetti/aspen_device.py

```diff
@@ -224,23 +225,23 @@ def __repr__(self):
         return f'cirq_rigetti.RigettiQCSAspenDevice(isa={self.isa!r})'
 
     def _json_dict_(self):
-        return {'isa': self.isa.to_dict()}
+        return {'isa': json.loads(self.isa.json())}
 
     @classmethod
     def _from_json_dict_(cls, isa, **kwargs):
-        return cls(isa=InstructionSetArchitecture.from_dict(isa))
+        return cls(isa=InstructionSetArchitecture.from_raw(json.dumps(isa)))
 
 
-@_provide_default_client  # pragma: no cover
+# pragma: no cover
 def get_rigetti_qcs_aspen_device(
-    quantum_processor_id: str, client: Optional[httpx.Client]
+    quantum_processor_id: str, client: Optional[QCSClient]
```

Comment:
Optional type needs to have a default value -
```suggestion
    quantum_processor_id: str, client: Optional[QCSClient] = None
```
Also please fix at
https://github.com/quantumlib/Cirq/blob/e36bb351ee98a9405dfcaf35227aa93e3a4f82a5/cirq-rigetti/cirq_rigetti/service.py#L116
and
https://github.com/quantumlib/Cirq/blob/e36bb351ee98a9405dfcaf35227aa93e3a4f82a5/cirq-rigetti/cirq_rigetti/service.py#L135